pkg/controller/controller.go

```diff
@@ -664,15 +663,23 @@ func (p *csiProvisioner) Delete(volume *v1.PersistentVolume) error {
 	storageClassName := volume.Spec.StorageClassName
 	if len(storageClassName) != 0 {
 		if storageClass, err := p.client.StorageV1().StorageClasses().Get(storageClassName, metav1.GetOptions{}); err == nil {
+
+			// Get PVC for secret reference if the volume is bound.
+			var pvc *v1.PersistentVolumeClaim
+			if volume.Spec.ClaimRef != nil {
+				pvc, _ = p.client.CoreV1().PersistentVolumeClaims(volume.Spec.ClaimRef.Namespace).Get(volume.Spec.ClaimRef.Name, metav1.GetOptions{})
```

Comment:
No need to check for this.  This will always fail because volume deletion is triggered by PVC being gone.